### PR TITLE
[store] in-memory storage engine

### PR DESCRIPTION
**Pull request by drmingdrmer**
## Summary

in-memory storage engine

## Changelog

- New Feature

### CODE REVIEW

**Comment by BohuTANG** [comment]
@drmingdrmer 

Hi xp,

error module is still refactoring(considering using anyhow instead), maybe the middle of this week can be done, I set this pr to draft mode(it means not ready for review).

**Comment by drmingdrmer** [comment]
@BohuTANG a in-memory engine as meta storage of flight service.

File: fusestore/store/src/api/rpc/flight_service.rs [hunk 1]
```diff
@@ -5,39 +5,56 @@
...
+use futures::Stream;
+use futures::StreamExt;
+#[allow(unused_imports)]
+use log::error;
+#[allow(unused_imports)]
 use log::info;
 use prost::Message;
 use tonic::metadata::MetadataMap;
-use tonic::{Request, Response, Status, Streaming};
+use tonic::Request;
```
Comment (sundy-li) [inline fusestore/store/src/api/rpc/flight_service.rs:35]: Try `cargo fmt`, this  will merge the imports

Comment (drmingdrmer) [reply fusestore/store/src/api/rpc/flight_service.rs:35]: I did. It does not merge `use` statement. And the one `use` per line style reduces conflicts with `git-merge` and `git-rebase`.

**Review by BohuTANG — APPROVED** [review]
LGTM

**Comment by BohuTANG** [comment]
Let's merge.
Thank you @drmingdrmer

File: fusestore/store/src/api/rpc/flight_service.rs (hunk 1 above)
Comment (BohuTANG) [reply fusestore/store/src/api/rpc/flight_service.rs:35]: We don't use merge group feature, but the style is group, it's fine here

**Pull request merged**